### PR TITLE
Restrict goal completion to live/viewable events

### PR DESCRIPTION
**Pull request by ethax-ross**
### Trello card

[Trello-3433](https://trello.com/c/3wu6pGzV/3433-investigate-why-historic-events-are-showing-up-in-ga-goal-completions)

### Context

At the moment its possible to reach the sign up `/completed` page for an event that is in the past/closed. As we have our goal completions setup to look for requests to this path its possible for a completion event to register in UA after the event
date (if a user has bookmarked the page or has left it open and reloads).

To prevent this we can redirect to the root event path, which will render a `410 gone` response (no longer registering as a goal completion). If the event is still 'live' but registrations are closed, the user will be presented with the event and closure message.

A 'walk in' can still technically sign up for a closed event (as this happensat the time of the event, so potenitally shortly after the start_at date). We need to make sure this is still possible by retaining the `is_walk_in`attribute in the store after sign up (otherwise the completion step will not know the sign up was a walk in and redirect them). 

### Changes proposed in this pull request

- Restrict goal completion to live/viewable events

- Use event factory in tests

As the controller action now inspects the date as well as the status of the event it needs to be present on our mock; we can do this easily by swapping it out to use the factory.

### Guidance to review

This PR also simplifies/changes the logic around determining if an event is 'viewable'. Previously we checked it was future dated and not pending. We should be safe to include closed in that conditional now as we no longer have a way of viewing past/archived events (previously you could view a past online Q&A event).

[Example completed page of past event](https://review-get-into-teaching-app-2644.london.cloudapps.digital/events/train-to-teach-london-event-050322/apply/completed)

### CODE REVIEW

**Comment by gemmadallmandfe** [comment]
@ethax-ross I can see this issue in prod - I've tried accessing a completed page for a historic event (`https://getintoteaching.education.gov.uk/events/train-to-teach-national-virtual-event-220622/apply/completed`)

However, I can't test in the review app as I don't know any URLs of historic events in the dev CRM! If you can let me know one, I'll test asap tomorrow

**Comment by gemmadallmandfe** [comment]
@misaka Can you have a look at this from a tech perspective so I can approve tomorrow once I've tested in the review app?

**Comment by gemmadallmandfe** [comment]
Sorry @ethax-ross - I missed the link for testing!

Tested successfully

**Review by gemmadallmandfe — APPROVED** [review]
Looks good to me

File: spec/requests/event_steps_controller_spec.rb [hunk 1]
```diff
@@ -179,11 +179,25 @@ def perform_request
   end
 
   describe "#completed" do
+    let(:path) { completed_event_steps_path(readable_event_id) }
```
Comment (misaka) [inline spec/requests/event_steps_controller_spec.rb:182]: Don't see aspecific reason why this is broken out into a `let`, is this a left-over from aborted changes below?

**Review by misaka — DISMISSED** [review]
LGTM!

File: spec/requests/event_steps_controller_spec.rb (hunk 1 above)
Comment (ethax-ross) [reply spec/requests/event_steps_controller_spec.rb]: Ah yeah I started doing a refactor of the specs above to consolidate the common setup but then decided against it. I've removed the let and fixed up the commit if you could re-tick it when you have a minute.

**Review by misaka — APPROVED** [review]

**Pull request merged**